mods/keys/craftitems.lua

```diff
@@ -0,0 +1,101 @@
+-- Load support for MT game translation.
+local S = minetest.get_translator("keys")
+
+--
+-- Craftitems
+--
+
+minetest.register_craftitem("keys:skeleton_key", {
+	description = S("Skeleton Key"),
+	inventory_image = "keys_key_skeleton.png",
+	on_use = function(itemstack, user, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			return itemstack
+		end
+
+		local pos = pointed_thing.under
+		local node = minetest.get_node(pos)
+
+		if not node then
+			return itemstack
+		end
+
+		local node_reg = minetest.registered_nodes[node.name]
+		local on_skeleton_key_use = node_reg and node_reg.on_skeleton_key_use
+		if not on_skeleton_key_use then
+			return itemstack
+		end
+
+		-- make a new key secret in case the node callback needs it
+		local random = math.random
+		local newsecret = string.format(
```

Comment:
This looks odd to me:

1. Why is this broken into 4 4-digit hex numbers instead of 2 8-digit ones?
2. Why is the one subtracted after the random? This might yield -1 if the random yields 0, which is likely to lead to very unexpected behavior:

```lua
assert(("%04x"):format(-1) == "ffffffffffffffff")
```

If you just want a random 8-byte string as a 16-byte hex string, why not use the following:

```lua
local newsecret = string.format("%08x%08x", random(0xFFFFFFFF), random(0xFFFFFFFF))
```

Reply:
This has been introduced in e4b1c935120580b013a6f9ba2212a4cb2ef246ac, and I did only move this code to a separate mod. I don't have answers to these 2 questions yet. I will look into this in detail later this week.